src/mtcp/mtcp_restart.c

```diff
@@ -975,6 +975,20 @@ read_one_memory_area(int fd)
     DPRINTF("WARNING: break (%p) not equal to end of heap (%p)\n",
             mtcp_sys_brk(NULL), area.addr + area.size);
   }
+  /* MAP_GROWSDOWN flag is required for stack region on restart to make
+   * stack grow automatically when application touches any address within the
+   * guard page region(usually, one page less then stack's start address).
+   *
+   * The end of stack is detected dynamically at checkpoint time. See
+   * prepareMtcpHeader() in threadlist.cpp and ProcessInfo::growStack()
+   * in processinfo.cpp.
+   */
+  if (area.name[0] && mtcp_strstr(area.name, "stack")
```

Comment:
Why do we compare the names here? Isn't endOfStack a more definitive way to find stack region?